### ISSUE TITLE
[Discussion] Show warning when breaking design rules

### ISSUE THREAD

**Issue by Zefau** [ioBroker/ioBroker.repochecker#84]
https://github.com/ioBroker/ioBroker.admin/issues/1772#issuecomment-1303038900
> From a developer perspective I would just appreciate certain rules and their enforcement. This topic is one of them. Another topic would be white spaces in ioBroker States / Objects.

> Jarvis rejects whitespaces but certain adapters create such states / objects.

> All adapters breaking such rules should at least be "punished" with a warning or error in the log. Wherever it comes from.

**Comment by Apollon77** [comment]
I transferred this to repo checker because we will have no way right now in js-controller

**Comment by Zefau** [comment]
still required from my perspective

**Comment by mcm1957** [comment]
Adapter checker cannot check which objects the adapter will creacte during execution. The ids of the objects can contain elements entered by the user (i.e. some names), elements defined by a device etc. Adapter developer should (better must) validate the ids i.e. by filtering using FORBIDDEN_CHARS. Bute thats nothing a staitc checker could do.

In addition js-controller replaces invalid chars as far as I know.

And as a side note: SPACES are not forbidden although strongly discouraged at stateIds. FORBIDDEN_CHARS allows spaces as far as I know.

In summary:
The static repository checker cannot check anything here..
